src/lib/provenanceUtils.ts

```diff
@@ -0,0 +1,16 @@
+import { ProvenanceEventTypes, State } from '@/types';
+import { createAction } from '@visdesignlab/trrack';
+
+export function updateProvenanceState(vuexState: State, label: ProvenanceEventTypes) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const stateUpdateActions = createAction<State, any[], ProvenanceEventTypes>((provState, newProvState) => {
+    if (label === 'Select Node' || label === 'De-select Node') {
+      provState.selectedNodes = newProvState.selectedNodes;
+    }
+  })
+    .setLabel(label);
+
+  if (vuexState.provenance !== null) {
+    vuexState.provenance.apply(stateUpdateActions(vuexState));
+  }
+}
```

Comment:
Can you help me understand what this function is doing?

Reply:
Yes, it takes in the new state (`vuexState`) and a label for the type of event, generates what the action function should be (in this case, just updating the selected nodes in the provenance state with the selected nodes from the new state), and applies that onto the provenance.

The utility of this action function syntax is that you can do more than just update the store. In other applications, here is where you'd do a d3 update, or something similar. Since our app has all the DOM logic handled by Vue, we just wanted to track the state changes, not apply any additional logic.